lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +856,81 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+/// Check if the input of this method (vector of NodeValues) can be
+/// legal input of CreateConcat function.
+/// \returns the dimension where the Concat will take place. If not
+/// concatenatable, return -1.
+static int getConcatDimensionForInputs(std::vector<NodeValue> inputs,
+                                       ConcatNode *origCN) {
+  auto firstDims = inputs.front().dims();
+  int dimSize = firstDims.size();
+  auto origCNInputShape = origCN->getInputs().front().dims();
+  int origCNDim = origCN->getDim();
+  int minimalNumOfUnitToLegallyConcat = 1;
+  // Find the minimal number of data units so that we can legally Concat them.
+  for (int i = origCNInputShape.size() - 1; i >= origCNDim; --i) {
+    minimalNumOfUnitToLegallyConcat *= origCNInputShape[i];
+  }
+
+  int index = -1;
+  int count = 0;
+  int preIndex = -1;
+  for (auto &in : inputs) {
+    auto curDims = in.dims();
+    // Find the previous non-negative index value.
+    preIndex = index == -1 ? preIndex : index;
+    count = 0;
+    index = -1;
+    for (int i = 0; i < dimSize; ++i) {
+      // Find the index i where firstDims[i] != curDims[i]
+      if (firstDims[i] != curDims[i]) {
+        count += 1;
```

Comment:
It seems like you should just be able to remove `count` entirely and just check if `index == -1` before setting it, and if not then return -1. Simplifies the code and readability.

Reply:
done